examples/synch-subscription.cpp

```diff
@@ -100,14 +140,24 @@ int main(int argc, char **argv) {
         std::cout << "subscription confirmed\n";
         rx_buff.consume(parse_result.consumed);
 
+        // endless receive loop
         while (true) {
+            // blocks until new message
             parse_result = c.read(rx_buff);
+            // extract (aka decouple from buffer / copy) result
             auto extract = boost::apply_visitor(r::extractor<Iterator>(),
                                                 parse_result.result);
+            // safe to consume buffer now
             rx_buff.consume(parse_result.consumed);
+
+            // select the result in the form of
+            // pair<const std::string&, const std::string&>
             auto payload = boost::apply_visitor(payload_extractor(), extract);
             if (payload) {
-                std::cout << "received payload:: '" << *payload << "'\n";
+                // and finally print it
+                std::cout << "on channel '" << payload->first
+                          << "' received payload :: '" << payload->second
+                          << "'\n";
```

Comment:
Really hard to understand this code! This is an example?

Also, I suggest `boost::string_ref` or `std::string_view` instead of `std::string const&`